### DISCUSSION TITLE
Need help. Setting key .jobname didn't work.

### DISCUSSION THREAD

**Discussion by Jjjjj-Lew** [James-Yu/LaTeX-Workshop#4299] [Q&A] [answered]
# Problem
I want to change the output file name and I tried to set the key ".jobname" but it didn't work. I have checked my output directory and found that the output file name as same as the input tex file name.
<img width="829" alt="image" src="https://github.com/James-Yu/LaTeX-Workshop/assets/124866979/f57f7e29-8dd1-4dc5-a5cd-c7fd94a05327">

# Enviroment
OS: Windows 11
VS Code 1.90.2
Tex compiler: MixTex

  I suspect this is caused by the different optional commands of MikTex, i.e. the "-job-name". Can some people help me figure it out? Other methods to customize output file name for individual project are also very welcome.

**Comment by James-Yu** [comment, marked as answer]
Add your jobname to your tool. `-jobname`.

**Reply by Jjjjj-Lew** [reply]
Thanks for your reply. I had thought to add the -jobname as an argument of my tool, but the setting is in the "User" scope, it will apply to all my latex projects and all the output files will be the same name if I forget to modify the setting. 
I might have misunderstood your point. Could you provide more details?

**Reply by Jjjjj-Lew** [reply]
Thank you once again. I edited the setting.json of the current workspace and added the "-job-name=manuscript" to the tool for this workspace, finally it works.
